brave-core/src/main/java/com/github/kristofa/brave/SpanIdGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.github.kristofa.brave;
+
+import com.github.kristofa.brave.internal.Nullable;
+
+import java.util.Random;
+
+/**
+ * Allows you to control how ids are generated. For example, implementations
+ * can encode where in the call tree the span originates from, or use alternate
+ * random number generators.
+ */
+// @FunctionalInterface
```

Comment:
This was intentional, but seems it failed to do the job ;) ex brave core is
java <8, but I want this type to be a functional interface so it can be
implemented with a lambda. I can make the note more explicit?